mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/MobileAcademyServiceImpl.java

```diff
@@ -51,99 +63,111 @@
+     * NMS course data service
      */
-    private CourseDataService courseDataService;
+    private NmsCourseDataService nmsCourseDataService;
 
     /**
      * Eventing system for course completion processing
      */
     private EventRelay eventRelay;
 
+    /**
+     * Used to retrieve course data
+     */
+    private SettingsFacade settingsFacade;
+
+    /**
+     * Used for alerting
+     */
+    private AlertService alertService;
+
     private static final Logger LOGGER = LoggerFactory.getLogger(MobileAcademyServiceImpl.class);
 
     @Autowired
-    public MobileAcademyServiceImpl(BookmarkDataService bookmarkDataService,
-                                    CourseDataService courseDataService,
+    public MobileAcademyServiceImpl(BookmarkService bookmarkService,
+                                    NmsCourseDataService nmsCourseDataService,
                                     CompletionRecordDataService completionRecordDataService,
-                                    EventRelay eventRelay) {
-        this.bookmarkDataService = bookmarkDataService;
-        this.courseDataService = courseDataService;
+
+                                    EventRelay eventRelay,
+                                    @Qualifier("maSettings") SettingsFacade settingsFacade,
+                                    AlertService alertService) {
+        this.bookmarkService = bookmarkService;
+        this.nmsCourseDataService = nmsCourseDataService;
         this.completionRecordDataService = completionRecordDataService;
         this.eventRelay = eventRelay;
+        this.settingsFacade = settingsFacade;
+        this.alertService = alertService;
+        bootstrapCourse();
     }
 
     @Override
-    public Course getCourse() {
+    public MaCourse getCourse() {
+
+        NmsCourse course = nmsCourseDataService.getCourseByName(COURSE_NAME);
+
+        if (course == null) {
+            alertService.create("mTraining.Course", COURSE_NAME, "Could not find course", AlertType.CRITICAL, AlertStatus.NEW, 0, null);
+            throw new IllegalStateException("No course bootstrapped. Check deployment");
+        }
 
-        // Make this course name configurable
-        String lookupName = "MobileAcademyCourse";
-        return courseDataService.findCourseByName(lookupName);
+        return mapCourseDomainToDto(course);
     }
 
     @Override
-    public void setCourse(Course course) {
-        Course existing = courseDataService.findCourseByName(course.getName());
+    public void setCourse(MaCourse courseDto) {
 
-        if (existing == null) {
-            courseDataService.create(course);
-        } else {
-            course.setId(existing.getId());
-            courseDataService.update(course);
+        if (courseDto == null) {
+            LOGGER.error("Attempted to set null course, exiting operation");
+            alertService.create("MA.Course", "MaCourse", "Trying to set null MaCourse", AlertType.CRITICAL, AlertStatus.NEW, 0, null);
```

Comment:
It looks like some of the alert subjects are MA.Course and some are mTraining.Course. Intentional?